### PR TITLE
Timlee/nice url paths

### PR DESCRIPTION
**Pull request by TimboTambo**
### Name of the story

[Github ticket](https://github.com/MHRA/products/issues/372)

### Testing information

See acceptance criteria.

### Pre-flight Checklist

- [ ] Tests
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

### CODE REVIEW

File: medicines/web/src/components/search-results/index.test.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const drugsMock = [
   },
 ];
 
-const noFeedback = () => {};
+const noFeedback = () => undefined;
```
Comment (rjkerrison) [inline medicines/web/src/components/search-results/index.test.tsx]: Why is this better than `() => {}` here?

File: medicines/web/src/components/search-results/pagination.tsx [hunk 2]
```diff
@@ -65,58 +65,54 @@ const StyledPagination = styled.nav`
...
+  const getHandlePageChangeHandler = (num: number) => () =>
+    props.handlePageChange(num);
+
   const createPaginationButton = (page: number, i: number, array: number[]) => {
     const separator = i === array.length - 1 ? '' : <span>&ndash;</span>;
 
     if (page === props.currentPage) {
       return (
-        <li key={page}>
+        <li key={page + i}>
```
Comment (rjkerrison) [inline medicines/web/src/components/search-results/pagination.tsx]: Is the `key` now `2 * page - 1`? Why?

File: medicines/web/src/components/search-results/pagination.tsx [hunk 3]
```diff
@@ -65,58 +65,54 @@ const StyledPagination = styled.nav`
...
     <StyledPagination>
       <ul className="pagination">
         {props.currentPage !== 1 ? (
-          <li className="arrow">
-            <Link href={getSearchPage(props.currentPage - 1)}>
-              <a>Previous</a>
-            </Link>
+          <li
+            className="arrow"
+            onClick={getHandlePageChangeHandler(props.currentPage - 1)}
```
Comment (rjkerrison) [inline medicines/web/src/components/search-results/pagination.tsx]: I don't think `li[onClick] > a` is great web semantics. If we can't use `Link` as we were before, should we remove the `a` tag here?

File: medicines/import/pars-dev-new-2020-02-24.log [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Running par
```
Comment (rjkerrison) [inline medicines/import/pars-dev-new-2020-02-24.log]: did you mean to commit these `.log` files?

File: medicines/import/pars-prod-new-2020-02-24.log [hunk 5]
```diff
@@ -0,0 +1,15 @@
+Running par
```
Comment (rjkerrison) [inline medicines/import/pars-prod-new-2020-02-24.log]: did you mean to commit these `.log` files?

File: medicines/web/src/components/page/index.tsx [hunk 6]
```diff
@@ -59,15 +59,17 @@ interface IPageProps {
...
-    });
-  }
+  useEffect(() => {
+    if (props.storageAllowed) {
+      TagManager.initialize({
+        gtmId: process.env.GOOGLE_GTM_CONTAINER_ID as string,
+        dataLayerName: 'dataLayer',
+      });
+      ReactGA.initialize(process.env.GOOGLE_TRACKING_ID as string, {
+        debug: true,
```
Comment (rjkerrison) [inline medicines/web/src/components/page/index.tsx]: do you want to use here the `process.env.GOOGLE_DEBUG` property you introduced into the pipeline?

File: medicines/web/src/components/search-results/index.test.tsx (hunk 1 above)
Comment (craiga) [reply medicines/web/src/components/search-results/index.test.tsx]: I had to add a `// tslint:disable-next-line: no-empty` to this in order for it to commit; maybe `undefined` works around this?

File: medicines/web/src/components/page/index.tsx [hunk 7]
```diff
@@ -58,16 +58,20 @@ interface IPageProps {
...
+    if (props.storageAllowed) {
+      TagManager.initialize({
+        gtmId: process.env.GOOGLE_GTM_CONTAINER_ID as string,
+        dataLayerName: 'dataLayer',
+      });
+      ReactGA.initialize(process.env.GOOGLE_TRACKING_ID as string, {
+        debug: useGoogleDebugMode,
+      });
+    }
+  }, [props.storageAllowed]);
```
[the remark this answers is not in the document]
Comment (TimboTambo) [reply medicines/web/src/components/page/index.tsx]: It cancels when the user toggles their preferences off, in the `cookie-form` component.
Then it should only be enabled again when the user accepts. Would be good to cover this in QA to backup my claim that this works as intended :P

File: medicines/web/src/components/search-results/pagination.tsx [hunk 8]
```diff
@@ -65,58 +65,54 @@ const StyledPagination = styled.nav`
...
+}
+
+const Pagination = (props: IPaginationProps) => {
   const pageCount = Math.floor(props.resultCount / props.pageSize) + 1;
   const { firstGroup, middleGroup, lastGroup } = getPaginationGroups(
     pageCount,
     props.currentPage,
   );
 
+  const getHandlePageChangeHandler = (num: number) => () =>
```
[the remark this answers is not in the document]
Comment (TimboTambo) [reply medicines/web/src/components/search-results/pagination.tsx]: Good point - removed this duplication.

File: medicines/web/src/components/search-results/pagination.tsx (hunk 3 above)
Comment (TimboTambo) [reply medicines/web/src/components/search-results/pagination.tsx]: True - was just being lazy as the styling was all about the `a`. Have replaced this now and also added some nav related tags for better accessibility

File: medicines/web/src/components/search-results/pagination.tsx (hunk 2 above)
Comment (TimboTambo) [reply medicines/web/src/components/search-results/pagination.tsx]: Was playing with the keys to silence the console error around unique keys as was trying to figure out if there were duplicate page number or something but it was actually complaining about the <li> below. Reverted this.

File: medicines/web/cypress/integration/tests.js [hunk 9]
```diff
@@ -68,6 +74,8 @@ const mockIbuprofenSpcPilResults = () =>
     'fixture:search_results.spcpil.json',
   );
 
+const longerTimout = 20000;
```
Comment (rjkerrison) [inline medicines/web/cypress/integration/tests.js]: I don't think we want a "longer Tim out"

File: medicines/web/src/pages/index.tsx [hunk 10]
```diff
@@ -15,31 +15,45 @@ const App: React.FC = () => {
...
     Events.viewPage('homepage');
   }, []);
 
   useEffect(() => {
-    if (search) {
-      const pageNum = page || 1;
+    // handle and redirect legacy urls
+    if (searchQS) {
+      const page = pageQS || 1;
+      const pathname = productQS ? '/product' : '/search';
```
Comment (rjkerrison) [inline medicines/web/src/pages/index.tsx]: just to confirm my understanding: this is here to reroute from the old query strings to the new pages?

**Pull request merged**